src/App.js

```diff
@@ -1,14 +1,29 @@
-import React from 'react';
+import React, {useCallback, useState} from 'react';
+import {Switch, BrowserRouter, Route, Redirect} from 'react-router-dom'
 import Header from 'pages/shared/Header/Header';
 import Footer from 'pages/shared/Footer/Footer';
 import Home from 'pages/Home/Home';
+import MovieDetails from "pages/MovieDetails/MovieDetails";
 
-const App = () => (
-    <>
-        <Header/>
-        <Home/>
-        <Footer/>
-    </>
-);
+const App = () => {
+
+    const [searchIconVisible, setShowSearchIconVisible] = useState(false);
+
+    const handleSearchIcon = (visibility) => {
+        setShowSearchIconVisible(visibility);
+    };
+
+    return (
+        <BrowserRouter>
+            <Header searchIconVisible={searchIconVisible} handleSearchIcon={handleSearchIcon}/>
+            <Switch>
+                <Route exact path="/:id" component={MovieDetails}/>
+                <Route exact path="/" render={() => <Home handleSearchIcon={handleSearchIcon}/>}/>
```

Comment:
I'm just curious can't we handle handleSearchIcon={handleSearchIcon} inside `Home` component? it doesn't seem that `const [searchIconVisible, setShowSearchIconVisible] = useState(false);` belongs to this component 🤔